### PR TITLE
Drop support for iOS 14 and remove availability checks

### PR DESCRIPTION
**Pull request by huwilkes**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Updated deployment target to iOS 15, and cleaned up availability checks.

### Verification

| Before | After |
|---|---|
| minOSBump_Button_Before | minOSBump_Button_After2 |
| minOSBump_CommandBar_Before | minOSBump_CommandBar_After |
| minOSBump_LargeContentViewer_Before | minOSBump_LargeContentViewer_After |
| minOSBump_Drawer_Before | minOSBump_Drawer_After |
| minOSBump_NavigationBar+BadeFieldButton_Before | minOSBump_NavigationBar+BadgeLabelButton_After |
| minOSBump_PillButtonBar_Before | minOSBump_PillButtonBar_After |
| minOSBump_SegmentedControl_Before | minOSBump_SegmentedControl_After |
| minOSBump_SideTabBar_Before | minOSBump_SideTabBar_After |
| minOSBump_TabBar_Before | minOSBump_TabBar_After |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1357)

### CODE REVIEW

File: ios/FluentUI.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1726,6 +1726,7 @@
 			baseConfigurationReference = 53FC90F525673626008A06FD /* FluentUI_release.xcconfig */;
 			buildSettings = {
 				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
Comment (harrieshin) [inline ios/FluentUI.xcodeproj/project.pbxproj:1729]: these changes required?

File: MicrosoftFluentUI.podspec [hunk 2]
```diff
@@ -12,7 +12,7 @@ Pod::Spec.new do |s|
 
 # iOS
 
-  s.ios.deployment_target = "14.0"
+  s.ios.deployment_target = "15.0"
```
Comment (harrieshin) [inline MicrosoftFluentUI.podspec:15]: i would expected our package.swift for SPM should be updated too?

File: MicrosoftFluentUI.podspec [hunk 3]
```diff
@@ -12,7 +12,7 @@ Pod::Spec.new do |s|
...
 
   s.subspec 'Avatar_ios' do |avatar_ios|
```
Comment (harrieshin) [inline MicrosoftFluentUI.podspec:17]: readme file should be updated too

File: ios/FluentUI.xcodeproj/project.pbxproj (hunk 1 above)
Comment (huwilkes) [reply ios/FluentUI.xcodeproj/project.pbxproj]: Looks like it'll just pull it from the xcconfig, so we're fine without these.

File: ios/FluentUI/Button/Button.swift [hunk 4]
```diff
@@ -188,9 +175,7 @@ open class Button: UIButton {
 
         if let image = image(for: .normal) {
             size.width += image.size.width
-            if #available(iOS 15.0, *) {
-                size.width += style.titleImagePadding
-            }
+            size.width += style.titleImagePadding
```
Comment (sophialee0416) [inline ios/FluentUI/Button/Button.swift:178]: Should we condense lines 177 + 178?

Comment (huwilkes) [reply ios/FluentUI/Button/Button.swift]: Turns out if we just flip the logic below, we can get rid of it

**Review by harrieshin — APPROVED** [review]

**Pull request merged**